### ISSUE TITLE
Pale Moon browser no longer works correctly with Habitica

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#10998]
There are two recent reports of Habitica not working correctly in the [Pale Moon](https://www.palemoon.org/) browser, which is "forked off from Firefox/Mozilla code a number of years ago... from mature Mozilla code and regularly updated". I'm going to copy the reports into their own comments here because they're a bit long and having them clearly separated might help. If I get time, I'll install the browser and do some testing of my own to see if I can get console logs, although one of the reporters has already tried that with no success.

The timing of the reports indicates that the problems probably started with release https://github.com/HabitRPG/habitica/releases/tag/v4.84.1 so I think this is a Habitica bug rather than a bug in Pale Moon.

**Comment by Alys** [comment]
(adding on to the end of my previous post)
... so I think this is a Habitica bug rather than a bug in Pale Moon.

**Comment by Alys** [comment]
@PixelsMom (f12552f2-05c8-4401-90b9-8ea621a5e4e2) reported in RaB 10 hours ago:

> Hi! Overnight, my clicks are failing to register (or are significantly more finicky in activating) when using the Pale Moon browser on the desktop version of Habitica. For example, clicking the checkbox dead center in my Tasks does nothing; I have to kind of blindly click around the area a few more times before something happens. Tasks, however, can be clicked, dragged and moved up and down the lists.
> 
> When I click the following, absolutely nothing happens: the Enchanted Armoire, User, Notifications, my Party's graphical icons in the header (I was trying to see what someone was wearing), food to feed my pet (while in the Stable), individual equipment pieces...Oh, and the three dots menu next to the tasks (to modify them).
> 
> Aside from that, I can click just about all the header words OK and they take me where I need to go ("Tasks, Inventory, Market, Groups, Help,") until I get to the gem image...everything to the right of that does not register my clicks.
> 
> It's really weird, it went from working perfectly fine for months in this browser to breaking overnight. The only significant change I've made is in signing up for a VPN the other day. But I've tried disconnecting from it and using Habitica and the results are the same.
> 
> I'm using Pale Moon browser 28.3.1 (it's an offshoot of Firefox). I have two Stylish extensions working: Collapsed Task Notes and Habitica Condensed Chats and one adblocker. I've tried restarting the browser in safe mode, logging out, clearing my Habitica cookies, logging back in...Trying after turning the VPN off, turning AdBlock Lite off, turning the Stylish extensions off.
> 
> However, I don't have this error using Epic Privacy Browser (which is Chromium-based, and I use no add-ons). Everything works as expected there, but I only use that as a backup browser. Any help sure would be appreciated! Thank you!

**Comment by Alys** [comment]
@shewhoknits (e1d43f13-4a7b-4188-8844-127b4f63c67c) first reported this a couple of days ago (after the release I mentioned above was released) and there's been an ongoing conversation about it in RaB:

SheWhoKnits:
> When I click the buttons "sync", "notifications" or "user", nothing happens. That means especially I can't log out, or is there another way?
> 
> When I logged in this morning, I had to press the "login" button (usually hitting "enter" is enough).
> 
> I'm using the desktop version and PaleMoon as a browser, but I don't think it depends on the browser because many other users seem to have similar problems.
> 
> ETA I can't "like" posts.

shanaqui: 
> I can't replicate this, and I can't actually see anyone else reporting these issues in the guild. Can you try going to [this link](https://habitica.com/static/clear-browser-data), clearing your browser data, and checking these things again?

SheWhoKnits:
> I tried again, cleared the Cache, logged out and logged in. To no avail, I can't even cross off tasks or anything else. Then I tried another browser (Edge) and I can do all the things so it does seem to be related to the browser (PaleMoon in my case). I guess the new website version no longer supports PaleMoon cry. I thought it was related to the other Bugs People reported this morning because the behavior was so similar, and it started exactly at the same time. (Sorry for wrong capitals and other mistakes, but the browser thinks it's German, and I haven't learned yet how to suppress autocorrect) BTW, in Edge I can't make the Chat box bigger."

Alys:
> Now that the recent bugs have been fixed, it's possible this will start working again in PaleMoon. Can you reload the website there and tell us if it still doesn't work correctly?

SheWhoKnits:
> I cleared the cache and tried again, but no, it no longer works cry. PaleMoon is a fork of Firefox and hasn't made the changes Firefox (Quantum) did. I had an addon in Firefox which stopped working with the new Firefox so I changed to PaleMoon where it still works. Some sites with very new technology don't work in PaleMoon, but so far I never needed those sites. But now I will have to change for Habitica. :(

Alys:
> Try these things to see if they work (it's important to do it in the order listed):
> 1. Disable all extensions even if they don't seem to be related (this is just temporary until we find out how we can make Habitica work again).
> 1. Go to https://habitica.com/static/clear-browser-data and click the red "Clear Data" button there, then BEFORE logging in again:
> 1. Reload the website using your browser's refresh button while you are holding down the "Control" or "Command" key on your keyboard.
> 1. After you've logged in, click the "sync" button. If it still doesn't work:
> 1. Open the Developer Console or JavaScript Console in your browser ([instructions here](https://webmasters.stackexchange.com/questions/8525/how-do-i-open-the-javascript-console-in-different-browsers)).
> 1. If you can see the trash can icon in the Console, click that (it's probably at the top left of console but don't worry about this step if you don't see it).
> 1. Click the sync button again.
> 1. If anything appears in the Console, paste it here (don't worry about neat formatting).

SheWhoKnits:
> I did that just now. I still had to press the "login" button (prior to the update I only needed to hit "enter" after giving my password). The sync button didn't work, and after step 7/8 the console window remained empty.
> 
> What I can do: follow links. Funny thing is that I get directly to this guild, but when I open "my guilds" ant then navigate to the STAG guild, I get the start window with tips for the website. <strike>But I can't check off tasks, hit any of the buttons below a chat box (like, report etc). The "show more" button on tasks doesn't work. I can create tasks with the gray box at the top, but not change them.</strike>
> 
> After writing the last sentence, I experimented some more with the green "+" button.
> 
> First I thought it didn't do a thing, but I clicked the mouse several times and suddenly a little menu opened. I didn't know how that happened, so I tried to replicate that. It looks like I have to hit a very special place to make it work. Same with editing tasks. I tried several places, and suddenly the edit window opened.
> 
> Then I tried the sync button again and found a little place where it suddenly worked.
> 
> My conclusion: I have only a very tiny area where mouse clicks work, but of course I can't see that area. That doesn't explain though why I get the start window when I navigate to the STAG guild.
> 
> Feel free to ask more questions :).

**Comment by Alys** [comment]
Note that SheWhoKnits (e1d43f13-4a7b-4188-8844-127b4f63c67c) has kindly offered to answer more questions, and I am guessing that PixelsMom (f12552f2-05c8-4401-90b9-8ea621a5e4e2) also wouldn't mind since they also took the time to make a detailed report (I'm about to ask them if that's okay and I'll post here if they say no). If we admins do have more questions, I'd suggest sending private messages so that both players don't have to keep checking back to RaB, and then copying the replies here.

**Comment by Alys** [comment]
I think this is actually caused by a recent update in one of the node modules that Habitica uses. I did some tests on my local install with older versions of the site. Here's the sequence of staps and results:

$ rm -rf node_modules/

$ git checkout ef412c7185a82f2e2b5a3fdc776a60380a35bebd
> commit ef412c7185a82f2e2b5a3fdc776a60380a35bebd
> Merge: 2eab8b2c8 c15b55808
> Author: Sabe Jones <sabrecat@gmail.com>
> Date:   Fri Feb 8 06:08:05 2019 -0600
> 	Merge branch 'release' into develop

$ npm install

\# the site works

$ git checkout 0155491a684be2d10442065e3b683624d9ce8134
>	commit 0155491a684be2d10442065e3b683624d9ce8134
>	Author: Matteo Pagliazzi <matteopagliazzi@gmail.com>
>	Date:   Sun Feb 10 19:03:35 2019 +0100
>		Upgrade vue (#10983)

\# site works

$ npm install

\# site doesn't work (no response when clicking on sync button, User Icon, etc)

Therefore something that got installed or upgraded from `package.json` in commit 0155491a684be2d10442065e3b683624d9ce8134 has caused the site to stop working in Pale Moon. I.e., there seems to be an incompatibility between Pale Moon and one of the third-party libraries that Habitica uses, rather than an incompatibility with Habitica's own code.

I suspect it might have something to do with a vue upgrade since these are the diffs between package.json in those two commits:
```
### "dependencies": 
14c14
<     "aws-sdk": "^2.329.0",
---
>     "aws-sdk": "^2.400.0",
65c65
<     "mongoose": "^5.3.4",
---
>     "mongoose": "^5.4.11",
101c101
<     "vue": "^2.5.16",
---
>     "vue": "^2.6.4",
106c106
<     "vue-template-compiler": "^2.5.16",
---
>     "vue-template-compiler": "^2.6.4",
### "devDependencies": 
147c147
<     "@vue/test-utils": "^1.0.0-beta.19",
---
>     "@vue/test-utils": "^1.0.0-beta.29",
```

However there are a lot of other differences between the package-lock.json files produced by the two npm install steps, so I can't be sure it's from vue.

**Comment by paglias** [comment]
It's very likely that this come from a change in Vue.js but unfortunately they don't support Pale Moon and we don't have the resources to support all existing browser apart from the major ones (Chrome, Firefox, Safari and IE/Edge).

I'll leave the issue open in case any contributor wants to propose a fix but we won't fix it ourselves. And it's possible that the necessary change will have to be done in Vue.js https://github.com/vuejs/vue and not in the Habitica code

**Comment by stefankeys** [comment]
Still not working....

**Comment by SabreCat** [comment]
Not a supported browser, so I'm going to close this for now.

**Comment by pachuco** [comment]
Confirming button issue with Basilisk browser, from the same guys that did PaleMoon. It's an FF v52 derivative.

The buttons fail to register click unless you fiddle with clicking around edge of button.
